fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt

```diff
@@ -200,10 +200,20 @@ class AddLoginFragment : Fragment(R.layout.fragment_add_login), MenuProvider {
                 override fun afterTextChanged(editable: Editable?) {
                     // update usernameChanged to true when the text is not empty,
                     // otherwise it is not changed, as this screen starts with an empty username.
-                    usernameChanged = editable.toString().isNotEmpty()
-                    updateUsernameField()
-                    setSaveButtonState()
-                    findDuplicate()
+                    when {
+                        editable.toString().isEmpty() -> {
+                            validUsername = false
+                            binding.clearUsernameTextButton.isVisible = false
+                        }
+                        else -> {
+                            usernameChanged = editable.toString().isNotEmpty()
```

Comment:
nit: editable.toString().isNotEmpty() and we know is false.
I think `usernameChanged` would always be `true` after the user interacted once with this field.